scrapy/downloadermiddlewares/httpcache.py

```diff
@@ -13,39 +16,58 @@
 from twisted.web.client import ResponseFailed
 
 from scrapy import signals
+from scrapy.crawler import Crawler
 from scrapy.exceptions import IgnoreRequest, NotConfigured
+from scrapy.extensions.httpcache import (
+    DbmCacheStorage,
+    DummyPolicy,
+    FilesystemCacheStorage,
+    RFC2616Policy,
+)
+from scrapy.http.request import Request
+from scrapy.http.response import Response
+from scrapy.settings import Settings
+from scrapy.spiders import Spider
+from scrapy.statscollectors import StatsCollector
 from scrapy.utils.misc import load_object
 
+TCrawler = TypeVar("TCrawler", bound=Crawler)
+TException = TypeVar("TException", bound=Exception)
+TRequest = TypeVar("TRequest", bound=Request)
+TResponse = TypeVar("TResponse", bound=Response)
+TSpider = TypeVar("TSpider", bound=Spider)
+TStatsCollector = TypeVar("TStatsCollector", bound=StatsCollector)
+
```

Comment:
Defining the types used in every current module is quite boring. There should be a better way to do this.

Reply:
I'm by no means a typing expert, but can't you just use the classes themselves? Like you are doing with the `settings` argument in `__init__`.

Reply:
The comment is adopted. Now the native class is used for typing hint.